global-hooks/calculate_resources_requests.go

```diff
@@ -76,118 +76,73 @@ var (
-
-		configEveryNodeMilliCPU int64
-		configEveryNodeMemory   int64
-
-		isManagedCloud bool
 	)
 	snapshots := input.Snapshots["NodesResources"]
-	if len(snapshots) > 0 {
-		// Hardcoded maximum values for master node resources
-		discoveryMasterNodeMilliCPU = hardLimitMilliCPU
-		discoveryMasterNodeMemory = hardLimitMemory
-		for _, snapshot := range snapshots {
-			n := snapshot.(*Node)
-			if n.AllocatableMilliCPU < discoveryMasterNodeMilliCPU {
-				discoveryMasterNodeMilliCPU = n.AllocatableMilliCPU
-			}
-			if n.AllocatableMemory < discoveryMasterNodeMemory {
-				discoveryMasterNodeMemory = n.AllocatableMemory
-			}
-		}
-	} else {
-		isManagedCloud = true
-		// Hardcoded maximum values for master nodes in managed clouds (GKE for example)
-		discoveryMasterNodeMilliCPU = managedHardLimitMilliCPU
-		discoveryMasterNodeMemory = managedHardLimitMemory
-	}
 
-	path := "global.modules.resourcesRequests.everyNode.cpu"
-	if !input.Values.Exists(path) {
-		return fmt.Errorf("%s must be set", path)
+	// Managed cloud
+	if len(snapshots) == 0 {
+		return nil
 	}
-	quantity, err := getAndParseResourceQuantity(input.Values.Get(path))
-	if err != nil {
-		return err
-	}
-	configEveryNodeMilliCPU = quantity.MilliValue()
-	if configEveryNodeMilliCPU <= 0 {
-		return fmt.Errorf("%s must be greater 0", path)
+
+	// Hardcoded maximum values for master node resources
+	discoveryMasterNodeMilliCPU = hardLimitMilliCPU
+	discoveryMasterNodeMemory = hardLimitMemory
+	for _, snapshot := range snapshots {
+		n := snapshot.(*Node)
+		if n.AllocatableMilliCPU < discoveryMasterNodeMilliCPU {
+			discoveryMasterNodeMilliCPU = n.AllocatableMilliCPU
+		}
+		if n.AllocatableMemory < discoveryMasterNodeMemory {
+			discoveryMasterNodeMemory = n.AllocatableMemory
+		}
 	}
 
-	path = "global.modules.resourcesRequests.everyNode.memory"
-	if !input.Values.Exists(path) {
-		return fmt.Errorf("%s must be set", path)
+	calculatedMasterNodeMilliCPU = discoveryMasterNodeMilliCPU - configEveryNodeMilliCPU
+	calculatedMasterNodeMemory = discoveryMasterNodeMemory - configEveryNodeMemory
+
+	if calculatedMasterNodeMemory <= 0 {
+		return fmt.Errorf("memory resources for allocating on master nodes must be greater than 0 (masterNode memory must be greater than 0 or discovered minimal master node memory must be greater than everyNode memory)")
 	}
-	quantity, err = getAndParseResourceQuantity(input.Values.Get(path))
-	if err != nil {
-		return err
+
+	if calculatedMasterNodeMilliCPU+configEveryNodeMilliCPU > discoveryMasterNodeMilliCPU {
+		return fmt.Errorf("everyNode CPU + masterNode CPU must be less than discovered minimal master node CPU")
 	}
-	configEveryNodeMemory = quantity.Value()
-	if configEveryNodeMemory <= 0 {
-		return fmt.Errorf("%s must be greater 0", path)
+
+	if calculatedMasterNodeMemory+configEveryNodeMemory > discoveryMasterNodeMemory {
+		return fmt.Errorf("everyNode memory + masterNode memory must be less than discovered minimal master node memory")
```

Comment:
I know, you did not change this in the PR. But the two last errors can never spawn. And this code branches are not covered with tests. I think it is worth refactoring as well as tests.